roles/flent/templates/workload.yml.j2

```
@@ -0,0 +1,92 @@
+---
+kind: Job
+apiVersion: batch/v1
+metadata:
+  name: 'flent-client-{{item.status.podIP}}-{{ trunc_uuid }}'
+  namespace: '{{ operator_namespace }}'
+spec:
+  template:
+    metadata:
+      labels:
+        app: flent-bench-client-{{ trunc_uuid }}
+        clientfor: {{ item.metadata.labels.app }}
+{% if workload_args.multus.enabled is sameas true %}
+      annotations:
+        k8s.v1.cni.cncf.io/networks: {{ workload_args.multus.client }}
+{% endif %}
+    spec:
+{% if workload_args.hostnetwork is sameas true %}
+      hostNetwork : true
+      serviceAccountName: benchmark-operator
+{% endif %}
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - {{ item.metadata.labels.app }}
+              topologyKey: kubernetes.io/hostname
+      containers:
+      - name: benchmark
+        image: {{ workload_args.image | default('quay.io/cloud-bulldozer/flent:latest') }}
+        env:
+          - name: uuid
+            value: "{{ uuid }}"
+          - name: test_user
+            value: "{{ test_user | default("ripsaw") }}"
+          - name: clustername
+            value: "{{ clustername }}"
+{% if elasticsearch is defined %}
+          - name: es
+            value: "{{ elasticsearch.url }}"
+          - name: es_index
+            value: "{{ elasticsearch.index_name | default("ripsaw-flent") }}"
+{% endif %}
+{% if flent.pin is sameas true %}
+          - name: client_node
+            client_node: {{ flent.pin_client }}
+          - name: client_node
+            server_node: {{ flent.pin_server }}
+{% endif %}
```

Adding client_node based on nodeName
```suggestion
          - name: client_node
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: server_node
            value: "{{ flent.pin_server|default("unknown") }}"
```